File: numbertree/submissions/accepted/numbertree_lukas.cpp
//Solution by lukasP (Lukáš Poláček)
#include <iostream>
#include <string>
using namespace std;

#define rep(i,a,b) for(__typeof(b) i=a; i<(b); ++i)

int main()
{
    int h;
    cin >> h;
    string s;
    cin >> s;

    int num = 1;
    rep(i,0,s.size())
        num = num * 2 + (s[i] == 'R');
    cout << ((1 << (h + 1)) - num) << endl;
}

File: numbertree/submissions/accepted/numbertree_lukas.cs
//Solution by lukasP (Lukáš Poláček)
using System;
using System.Linq;

class Numbertree
{
    public static void Main()
    {
        string[] ss = Console.ReadLine().Split();
        int h = int.Parse(ss[0]);
        int num = ss[1].Aggregate(1, (x, y) => 2 * x + Convert.ToInt32(y == 'R'));
        Console.WriteLine((1 << (h + 1)) - num);
    }
}
